cmd/operator/test/user.go

```diff
@@ -0,0 +1,80 @@
+package test
+
+import (
+	"context"
+	"math/big"
+	"net/rpc"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	operator "github.com/perun-network/erdstall/cmd/operator"
+	"github.com/perun-network/erdstall/contracts/bindings"
+	"github.com/perun-network/erdstall/tee"
+)
+
+type User struct {
+	*testing.T
+	ethClient *operator.EthereumClient
+	rpcClient *rpc.Client
+	contract  *bindings.Erdstall
+	nonce     uint64
+}
+
+const gasLimit = 2000000
+const defaultContextTimeout = 10 * time.Second
+
+func newDefaultContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), defaultContextTimeout)
+}
+
+func CreateUser(t *testing.T, ethUrl string, privateKey string, rpcUrl string, contractAddress common.Address) *User {
+	ethClient, err := operator.CreateEthereumClient(ethUrl, privateKey)
+
+	rpcClient, err := rpc.DialHTTP("tcp", rpcUrl)
+	if err != nil {
+		t.Fatal("dialing rpc:", err)
+	}
+
+	contract, err := bindings.NewErdstall(contractAddress, ethClient)
+	if err != nil {
+		t.Fatal("loading contract:", err)
+	}
+
+	return &User{t, ethClient, rpcClient, contract}
+}
+
+func (u *User) Deposit(amount int64) {
+	ctx, cancel := newDefaultContext()
+	defer cancel()
+
+	t, err := u.ethClient.NewKeyedTransactor(ctx, gasLimit)
+	if err != nil {
+		u.Fatal("creating keyed transactor:", err)
+	}
+
+	t.Value = new(big.Int).SetInt64(amount)
+
+	u.contract.Deposit(t)
```

Comment:
don't forget to `bind.WaitMined` after you send any tx.

Reply:
good point